### PR TITLE
feat(carousel-card-section-offset): add print styles to Carousel and Card Section Offset

### PR DESCRIPTION
**Pull request by mwanberg**
### Related Ticket(s)

Github: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7414

### Description

"This request originally stemmed from discussions with the case study team and the Afterburner experience guild team as we collaborated on the new case study article template. We wanted the Carbon for ibm.com components to be able to be assembled together in a way that when the user chooses to print the web page, it is visually aligned with what the components should look like. Specifically for case study team, this will allow the team to create custom PDFs for the more important case studies and rely on the print-friendly view for all other case studies. This use case seems applicable not just for the case study team, but for all other pages that uses the Carbon components (e.g. Learn, Solution)."

### Changelog

**New**

- Print styles for the Carousel and Card Offset Section

### CODE REVIEW

**Comment by proeung** [comment]
@mwanberg Is this PR almost ready for the "Review" status?

**Comment by mwanberg** [comment]
@proeung Yes, this is ready for review!

**Review by IgnacioBecerra — COMMENTED** [review]
Looking good @mwanberg! The Card Section offset should look good with the suggested changes (tested with Ctrl+P). As for the Carousel, not sure what changed from the draft PR since now the Cards take up full width, could you take another look? 🤔 

Feel free to reach out if you have any more questions!

File: packages/styles/scss/components/carousel/_carousel.scss [hunk 1]
```diff
@@ -99,3 +99,44 @@
...
+
+      &:hover {
+        background-color: transparent;
+      }
+    }
+
+    .#{$prefix}--tile {
+      &:hover {
+        background-color: transparent;
+      }
```
Comment (proeung) [inline packages/styles/scss/components/carousel/_carousel.scss:139]: I don't think we need to define a hover state style for print.

**Review by proeung — DISMISSED** [review]
Looks good, however, there's some code refinement that's needed. Please take a look at @IgnacioBecerra's feedback.

File: packages/web-components/src/components/card-group/card-group.scss [hunk 2]
```diff
@@ -98,3 +98,35 @@
     }
   }
 }
+
+// Print styles
+@media print {
```
[the remark this answers is not in the document]
Comment (mwanberg) [reply packages/web-components/src/components/card-group/card-group.scss:103]: I cleaned up a few of the lines, but the ones I left in are overriding styles that are in that file. If I take them out, the other styles in this file takes precedence over the ones under the `styles` folder. Let me know if there's a better way to handle this!

**Comment by RichKummer** [comment]
The styles are looking good on the design side. I did notice that when trying to print for `Card section offset`, the card heading and body content overlap the CTA button and the border of the card. `Carousel` doesn't look like it's having the same issue. Any way we could fix this for `Card section offset`?
Screen Shot 2021-11-18 at 2 38 25 PM

**Comment by mwanberg** [comment]
> The styles are looking good on the design side. I did notice that when trying to print for `Card section offset`, the card heading and body content overlap the CTA button and the border of the card. `Carousel` doesn't look like it's having the same issue. Any way we could fix this for `Card section offset`? 

Thank you for reviewing this! It looks like the font sizes might need to be stepped down to make them fit appropriately in the card. Currently they're set using the variables specified in the specs; should I adjust that down anyway? Also, I found there are browser differences, too, in how they each interpret print styles.

@IgnacioBecerra @proeung what do you two think? Should I try to edit the cards' font sizing to get them to better approximate the specs, even though it won't be the same specified typography variables?

**Comment by proeung** [comment]
@RichKummer @mwanberg The text overlapping issue seen within the actual print mode looks to be coming from the `sameHeight()` function where it's not resizing the card based on the page layout. If the user sets the layout to "Landscape", it renders the cards just fine. @IgnacioBecerra Feels like this is an issue that will need to be accounted for in a separate ticket so that we can evaluate the correct solution and keep this one to just print styles. Let me know what you think.

Screen Shot 2021-11-22 at 12 27 37 PM

**Comment by RichKummer** [comment]
Hey @proeung, I was able to go in and replicate. I think that solution works, we can write a follow-up ticket to cover the text overflowing issue. Thank you for looking into this!

**Review by RichKummer — APPROVED** [review]

**Comment by proeung** [comment]
@RichKummer Sounds good and no problem!

**Comment by mwanberg** [comment]
@IgnacioBecerra @proeung I moved the print styles completely into the `styles` package, though I did have to use `!important`. :( 

(I copied what I saw in the footer print styles.)

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, thanks for addressing all these comments @mwanberg! You just gotta comment `I have read the DCO document and I hereby sign the DCO.` so this is able to be merged :)

**Comment by jeffchew** [comment]
@proeung this one looks like is good to go, are there additional changes needed?

@mwanberg could you accept the DCO? That is the last required check here.

**Comment by mwanberg** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by jeffchew** [comment]
recheck

**Pull request merged**